port/common/omrportcontrol.c

```diff
@@ -289,6 +289,13 @@ omrport_control(struct OMRPortLibrary *portLibrary, const char *key, uintptr_t v
 		return 0;
 	}
 
+	if (0 == strcmp(OMRPORT_CTLDATA_VMEM_ADVISE_HUGEPAGE, key)) {
+#if defined(LINUX)
+		/* set value to advise OS about vmem to consider for Transparent HugePage (Only for Linux) */
+		portLibrary->portGlobals->vmemEnableMadvise &= value;
```

Comment:
Why is the value being `&` and not just directly stored?

Reply:
`&` is to make sure that `MADV_HUGEPAGE` flag is only set when THP setting is `[madvise]`, there is not point calling madvise for other cases even if user specified to enable this.